SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialGameInstance.cpp

```diff
@@ -237,26 +240,33 @@ void USpatialGameInstance::HandleOnPlayerSpawnFailed(const FString& Reason)
 	OnSpatialPlayerSpawnFailed.Broadcast(Reason);
 }
 
-void USpatialGameInstance::OnLevelInitializedNetworkActors(ULevel* LoadedLevel, UWorld* OwningWorld)
+void USpatialGameInstance::CleanupLevelInitializedNetworkActors() const
```

Comment:
I'd be great if this could just operate on the loaded level, could we cache them if they are loaded before connection, and parse them on connection? Otherwise we may way to validate that the actors we're operating on are actual level actors.

Reply:
I was really trying to get rid of that callback entirely and use the MapLoaded callback, but I couldn't find a way to do that and also this, so for now I'm just cache'ing the Levels and then processing them when we connect. It does mean this code has to live in the GameInstance for the foreseeable future, since the NetDriver isn't created early enough.